src/main/java/edu/ucsb/cs156/courses/services/UCSBCurriculumService.java

```diff
@@ -166,4 +166,41 @@ public String getSectionJSON(String quarter, String enrollCode) {
         logger.info("json: {} contentType: {} statusCode: {}",retVal,contentType,statusCode);
         return retVal;
     }
-}
\ No newline at end of file
+
+  public static final String FINAL_ENDPOINT = "https://api.ucsb.edu/academics/curriculums/v1/finals";
+
+  public String getFinalJSON(String quarter, String enrollCode) {
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON));
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set("ucsb-api-version", "1.0");
+        headers.set("ucsb-api-key", this.apiKey);
+
+        HttpEntity<String> entity = new HttpEntity<>("body", headers);
+
+        String params = String.format(
+            "?quarter=%s&enrollCode=%s", quarter, enrollCode);
+        String url = FINAL_ENDPOINT + params;
+
+        logger.info("url=" + url);
+
+        String retVal = "";
+        MediaType contentType=null;
+        HttpStatus statusCode=null;
+        try {
+            ResponseEntity<String> re = restTemplate.exchange(url, HttpMethod.GET, entity, String.class);
+            contentType = re.getHeaders().getContentType();
+            statusCode = re.getStatusCode();
+            retVal = re.getBody();
+        } catch (HttpClientErrorException e) {
+            retVal = "{\"error\": \"401: Unauthorized\"}";
+        }
+        if (retVal.contains("null")){
+            retVal = "{\"error\": \"No final exam found, contact professor for details.";
+        }
```

Comment:
Ooops... syntax error in the JSON:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/1119017/172243369-86d9deec-18a8-41b9-8a7e-c3e151d2c842.png">


A better approach is described here:
https://stackoverflow.com/a/29340696

In this case, something like: 

```java
Map<String,String> payload = new HashMap<>();
payload.put("error","No final exam found; contact professor for details");
retVal = new ObjectMapper().writeValueAsString(payload);
```

Reply:
implemented changes for json syntax error